packages/11ty/_plugins/filters/sortByKeys.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * 
+ * Sort comparator using the keys in the ordered array
+ * All keys will be sorted in the `sortOrder` passed unless a key specific order
+ * is set in the SortKey item.
+ *
+ * @typedef SortKey
+ * @type (Array|Object)
+ * @property {String} key - key name
+ * @property {String} order ["ascending"|"descending"] - sort order for key
+ *
+ * @example
+ *   ['priority', 'author', 'title'], 'ascending']
+ * @example
+ *   [['priority', 'ascending'], ['date','descending']]
+ * @example
+ *   [{ key:'priority', order:'ascending' }, { key:'date', order:'descending' }]
+ *
+ * @param  {Array<SortKey>}  sortKeys - key name and optional key specific sort order
+ * @param  {String}  sortOrder - Default sort order, "ascending" or "descending"
+ *
+ * @return  {Number}  Comparison 1, 0, -1
+ */
+module.exports = (sortKeys = [], sortOrder = 'ascending') => {
+  if (!sortKeys.length) return
+
+  if (!['ascending', 'descending'].includes(sortOrder)) {
+    throw new Error(`
+
+      Invalid sort order "${sortOrder}" passed to sortByKeys,\n
```

Comment:
If the additional empty line is intentional we might instead use a new line character, `\n`, like so
```suggestion
    throw new Error(`\n
      Invalid sort order "${sortOrder}" passed to sortByKeys,\n
```